### PR TITLE
Feature/gqss 4217 baremetal port setup nera

### PR DESCRIPTION
**Pull request by nnesic**

### CODE REVIEW

File: core/src/com/cloud/agent/resource/virtualnetwork/model/BaremetalRules.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.cloud.agent.resource.virtualnetwork.model;
```
Comment (HrWiggles) [inline core/src/com/cloud/agent/resource/virtualnetwork/model/BaremetalRules.java]: Missing license comment

File: server/src/com/cloud/network/router/CommandSetupHelper.java [hunk 2]
```diff
@@ -487,17 +463,25 @@ public void createFirewallRulesCommands(final List<? extends FirewallRule> rules
...
+    }
+
+    public void createBaremetalPxeRulesCommands(final List<? extends FirewallRule> rules, final VirtualRouter router, final Commands cmds) {
+        final List<FirewallRuleTO> rulesTO = new ArrayList<FirewallRuleTO>();
+        String systemRule = null;
+        Boolean defaultEgressPolicy = false;
+        if (rules != null) {
+
+            for (final FirewallRule rule : rules) {
+                final FirewallRule.TrafficType traffictype = rule.getTrafficType();
```
Comment (HrWiggles) [inline server/src/com/cloud/network/router/CommandSetupHelper.java]: Unused variable `traffictype`

File: server/src/com/cloud/network/router/CommandSetupHelper.java [hunk 3]
```diff
@@ -1011,7 +940,8 @@ private VmDataCommand generateVmDataCommand(final VirtualRouter router, final St
 
         cmd.addVmData("metadata", "local-ipv4", guestIpAddress);
         cmd.addVmData("metadata", "local-hostname", StringUtils.unicodeEscape(vmName));
-        if (dcVo.getNetworkType() == NetworkType.Basic) {
+        // TODO see this
```
Comment (HrWiggles) [inline server/src/com/cloud/network/router/CommandSetupHelper.java]: Should this comment be here?

File: systemvm/patches/debian/config/opt/cloud/bin/configure.py [hunk 4]
```diff
@@ -910,6 +910,70 @@ def setupMetadataRules(self):
...
+            if "icmp_type" in rule and rule['icmp_type'] != -1:
+                icmp_type = rule['icmp_type']
+            if "icmp_code" in rule and rule['icmp_code'] != -1:
+                icmp_type = "%s/%s" % (icmp_type, self.rule['icmp_code'])
+            return '--icmp-type %s' % icmp_type
+        else:
+            if 'src_port_range' in rule:
+                first_port = rule['src_port_range'][0]
+                last_port = rule['src_port_range'][1]
+                rnge = ''
```
Comment (HrWiggles) [inline systemvm/patches/debian/config/opt/cloud/bin/configure.py]: Unused variable `rnge`

File: systemvm/patches/debian/config/opt/cloud/bin/configure.py [hunk 5]
```diff
@@ -1000,7 +1068,10 @@ def main(argv):
 
         acls = CsAcl('firewallrules', config)
         acls.process()
-
+       
+        bm = CsBaremetalRules('baremetalrules', config)
```
Comment (HrWiggles) [inline systemvm/patches/debian/config/opt/cloud/bin/configure.py]: Was it not possible to simply use `CsAcl`? (instead of adding new data bag `CsBaremetalRules`)
Would it not have been possible to simply do
```
        bm = CsAcl('baremetalrules', config)
        bm.process()
```
?

Comment (nnesic) [inline systemvm/patches/debian/config/opt/cloud/bin/configure.py]: I looked into that, the problem is it tries to list the rules under FIREWALL_something chain, rather than just input chain.

Comment (nnesic) [inline systemvm/patches/debian/config/opt/cloud/bin/configure.py]: Also, it generates a bunch of additional rules that we don't need in mangle. I was thinking in the future to refactor this as "Router Supported Services", where we'd allow network offerings to specify router-specific rules, which could then use the new class.

File: server/src/com/cloud/network/router/CommandSetupHelper.java (hunk 3 above)
Comment (nnesic) [inline server/src/com/cloud/network/router/CommandSetupHelper.java]: Ah yes, actually I wanted to ask you about it - I saw the change in the part of code you did, from "if (dcVo.getNetworkType() == NetworkType.Basic)" to "if (NetworkType.valueOf(cmd.getAccessDetail(NetworkElementCommand.ZONE_NETWORK_TYPE)) == NetworkType.Basic) ". I was wondering why did you change it?

Comment (HrWiggles) [inline server/src/com/cloud/network/router/CommandSetupHelper.java]: Since I had refactored things so that common access details got set by `setCommonNetworkElementCommandAccessDetails` (instead of having the same code all over the place), there was no longer a `dcVo` here.
So that left two choices: re-add `dcVo` here (resulting in a second call to `_dcDao.findById`) or simply reading the zone type from the access details that have just been set by call to `setCommonNetworkElementCommandAccessDetails`.
I chose the option that I felt would require the least amount of processing time/power.

Comment (nnesic) [inline server/src/com/cloud/network/router/CommandSetupHelper.java]: Aha, I see.
